code/modules/mob/living/carbon/xenomorph/castes/carrier/carrier.dm

```diff
@@ -17,12 +17,49 @@
 	inherent_verbs = list(
 		/mob/living/carbon/xenomorph/proc/vent_crawl,
 	)
+	///Facehuggers overlay
+	var/mutable_appearance/hugger_overlays_icon
 
 // ***************************************
 // *********** Life overrides
 // ***************************************
-/mob/living/carbon/xenomorph/carrier/Stat()
+/mob/living/carbon/xenomorph/carrier/Initialize(mapload)
 	. = ..()
+	hugger_overlays_icon = mutable_appearance('icons/Xeno/2x2_Xenos.dmi',"empty")
 
+/mob/living/carbon/xenomorph/carrier/Stat()
+	. = ..()
 	if(statpanel("Game"))
 		stat("Stored Huggers:", "[huggers] / [xeno_caste.huggers_max]")
+
+/mob/living/carbon/xenomorph/carrier/update_icons()
+	. = ..()
+	update_hugger_overlays()
+
+/mob/living/carbon/xenomorph/carrier/proc/update_hugger_overlays()
```

Comment:
Why are you making a separate proc? Just use the existing update_overlays proc.

Reply:
Done?

Reply:
Update_icons already calls update_overlays by default, so you don't have to specifically call it. This is part of why its important to use that proc as it's already built in.